openmdao/core/component.py

```diff
@@ -410,7 +408,7 @@ def add_output(self, name, val=1.0, shape=None, units=None, res_units=None, desc
         dict
             metadata for added variable
         """
-        if inspect.stack()[1][3] == '__init__':
+        if self._static_mode:
```

Comment:
actually we're ok here because IndepVarComps.add_output just stores the args and then later, during setup, calls the base class version of add_output.